src/web-eid.ts

```diff
@@ -78,72 +98,119 @@ export async function status(): Promise<Versions> {
+  await extensionLoadDelay();
 
-  if (typeof options != "object") {
-    throw new MissingParameterError("authenticate function requires an options object as parameter");
+  if (!challengeNonce) {
+    throw new MissingParameterError("authenticate function requires a challengeNonce");
   }
 
-  if (!options.getAuthChallengeUrl) {
-    throw new MissingParameterError("getAuthChallengeUrl missing from authenticate options");
-  }
+  const timeout = (
+    config.EXTENSION_HANDSHAKE_TIMEOUT +
+    config.NATIVE_APP_HANDSHAKE_TIMEOUT +
+    (options?.userInteractionTimeout || config.DEFAULT_USER_INTERACTION_TIMEOUT)
+  );
 
-  if (!options.postAuthTokenUrl) {
-    throw new MissingParameterError("postAuthTokenUrl missing from authenticate options");
-  }
+  const message: ExtensionAuthenticateRequest = {
+    action:         Action.AUTHENTICATE,
+    libraryVersion: config.VERSION,
+
+    challengeNonce,
+    options,
+  };
+
+  const {
+    unverifiedCertificate,
+    algorithm,
+    signature,
+    format,
+    appVersion,
+  } = await webExtensionService.send<ExtensionAuthenticateResponse>(message, timeout);
+
+  return {
+    unverifiedCertificate,
+    algorithm,
+    signature,
+    format,
+    appVersion,
+  };
+}
+
+export async function getSigningCertificate(options?: ActionOptions): Promise<LibraryGetSigningCertificateResponse> {
+  await extensionLoadDelay();
 
   const timeout = (
     config.EXTENSION_HANDSHAKE_TIMEOUT +
     config.NATIVE_APP_HANDSHAKE_TIMEOUT +
-    (options.serverRequestTimeout || config.DEFAULT_SERVER_REQUEST_TIMEOUT) * 2 +
-    (options.userInteractionTimeout || config.DEFAULT_USER_INTERACTION_TIMEOUT)
+    (options?.userInteractionTimeout || config.DEFAULT_USER_INTERACTION_TIMEOUT) * 2
   );
 
-  const message = {
-    ...options,
-
-    action:         Action.AUTHENTICATE,
+  const message: ExtensionGetSigningCertificateRequest = {
+    action:         Action.GET_SIGNING_CERTIFICATE,
     libraryVersion: config.VERSION,
+
+    options,
   };
 
-  const result = await webExtensionService.send<ResponseAuthenticateSuccess>(message, timeout);
+  const {
+    certificate,
+    supportedSignatureAlgorithms,
+  } = await webExtensionService.send<ExtensionGetSigningCertificateResponse>(message, timeout);
 
-  return result.response;
+  return {
+    certificate,
+    supportedSignatureAlgorithms,
+  };
 }
 
-export async function sign(options: SignOptions): Promise<HttpResponse> {
-  await defer(); // Give chrome a moment to load the extension content script
+export async function sign(
+  certificate: string,
+  hash: string,
+  hashFunction: string,
+  options?: ActionOptions,
+): Promise<LibrarySignResponse> {
+  await extensionLoadDelay();
 
-  if (typeof options != "object") {
-    throw new MissingParameterError("sign function requires an options object as parameter");
+  if (!certificate) {
+    throw new MissingParameterError("sign function requires a certificate as parameter");
   }
```

Comment:
This could be more DRY here and elsewhere:

```suggestion
  requireParameter(sign, "certificate", certificate);
```
with the following implementation:
```js
function requireParameter(func, parameterName, parameterValue) {
  if (!parameterValue) {
    throw new MissingParameterError(`${func.name}() requires a non-empty ${parameterName} as parameter`);
  }
}
```